### DISCUSSION TITLE
Enablement of discussions for other repositories

### DISCUSSION THREAD

**Discussion by MikeMcC399** [corona-warn-app/cwa-website#3282] [General] [answered]
Following on from https://github.com/corona-warn-app/cwa-website/discussions/3279#discussioncomment-4326526

@brianebeling

> In regards to enabling discussions on other repos, I'd be careful with that. As discussed in corona-warn-app/cwa-wishlist#360 there are some downsides. For the Wishlist it would make a lot of sense to enable it in my opinion. For something like cwa-server it might not be as helpful. But that depends on the people that actually maintain it. I for one think its helpful to differentiate between general conversation and actionable items. You can also always convert a discussion to an issue if it becomes actionable or vice versa.

The CWA GitHub repositories are a mixture of different usages. 

There are classical GitHub repositories where content in the repository is maintained through issues, commits and pull requests e.g.

- https://github.com/corona-warn-app/cwa-website (this repository)
- https://github.com/corona-warn-app/cwa-server 
- https://github.com/corona-warn-app/cwa-app-android
- https://github.com/corona-warn-app/cwa-app-ios

Then there is the documentation repository where own documentation content is managed and additionally issues which affect both https://github.com/corona-warn-app/cwa-app-android and https://github.com/corona-warn-app/cwa-app-ios are maintained:

- https://github.com/corona-warn-app/cwa-documentation

Finally there are unconventional repositories where no content is managed and only issue lists are maintained:

- https://github.com/corona-warn-app/cwa-wishlist
- https://github.com/corona-warn-app/cwa-hotline

This is not an exhaustive list. The full list is on https://github.com/orgs/corona-warn-app/repositories.

- https://github.com/corona-warn-app/cwa-server already has a discussion list, however there has been no posting there for over one year.

In the meantime, the issue lists of https://github.com/corona-warn-app/cwa-app-android and https://github.com/corona-warn-app/cwa-app-ios are showing little activity.

https://github.com/corona-warn-app/cwa-documentation has slow activity, however there are some entries from @Ein-Tim which were questions. These might be better handled by enabling a discussion list there.

The https://github.com/corona-warn-app/cwa-wishlist is already just a list of suggestions with no other repository content. Adding a discussion list there would make it more confusing. If anything I would direct questions to the documentation repository.

In general limiting the number of lists which need to be monitored, managed and maintained has its benefits. When searching for items I know exist I regularly have problems trying to find them due to the spread of topics over the repositories, even though it is possible to search in the organisation. So there should be clear benefits if any other discussion lists are added to other repositories.

**Comment by Ein-Tim** [comment]
I am still convinced from the answer @heinezen gave in https://github.com/corona-warn-app/cwa-wishlist/issues/360 ans think that enabling discussions in general is not a good idea, as it spreads out the lists where (potentially important) content is talked about. 

So I vote for: **Do not enable discussions on more repositories!**

**Comment by brianebeling** [comment, marked as answer]
Thank you for the write up and the summary. First and foremost, discussions won't be enabled in other repositories (at least right now). It just made a lot of sense for the ticket about the badges. It was neither code related, nor mirrored to an internal ticket, nor on any task board. It could be answered, but was not actionable. Ultimately it was a nice suggestion, but we can't close it as "done" and can't really force all individuals to do that. Even though it is helpful, indeed. 

If anything I'd argue it makes it easier to navigate the repository. I'm not saying it makes it easier to navigate the organization though.

Really it depends on how the maintainers work.
If a difference between an issue and a generic discussion can be made, then it's easier to find what you are looking for. In reality 95% of user created content will still be an issue. Only a handful will most likely be a discussion. Not exact numbers, I'm just trying to illustrate the dynamic. That has an important benefit though. These few discussions are most likely valuable and helpful for others. 

I'd also say that it has another less obvious implication. Usually issues are mirrored by the cwa-bot. Sometimes that is not necessary however and it pollutes the board. Then on the other hand sometimes it is necessary and it doesn't happen.

Well, in any case. I have more thoughts regarding this, but I'm traveling at the moment and typing on mobile takes a while.

Rest assured not much will change. Maybe it won't even be used and we can disable it. Right now I'd treat it like an experiment and see what happens. If it doesn't work out we can just convert to issues anyways.

For me it's helpful to have less noise in issues. And for others it might be helpful to use polls or the "select as answer" option. Discussions will of course also be moderated, just like the issues.

**Reply by Ein-Tim** [reply]
> Usually issues are mirrored by the cwa-bot. Sometimes that is not necessary however and it pollutes the board.

I have the feeling these boards aren't really maintained anymore from CM side (e.g. removing closed issues from the "To Do" list, if the bot for some reason moved the issue to "To Do" but there is actually nothing left to do), or am I wrong here?

**Comment by Ein-Tim** [comment]
One positive thing about discussions: There is a "Show and Tell" option, which @MikeMcC399 used in #3283. That is actually useful, as we quite often had the situation in this repository where issues were opened with remarks, but there was nothing that needed fixing/changes instantly.
